prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/repository/NodesClient.java

```diff
@@ -0,0 +1,123 @@
+/*-
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.prediction_applier.repository;
+
+import static org.apache.camel.Exchange.HTTP_RESPONSE_CODE;
+
+import java.net.UnknownHostException;
+import java.util.Set;
+
+import com.fasterxml.jackson.core.io.JsonEOFException;
+import com.fasterxml.jackson.databind.exc.MismatchedInputException;
+import lombok.RequiredArgsConstructor;
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.apache.hc.client5.http.HttpHostConnectException;
+import org.apache.hc.core5.http.MalformedChunkCodingException;
+import org.apache.hc.core5.http.NoHttpResponseException;
+import org.springframework.stereotype.Component;
+
+import org.alfresco.hxi_connector.common.exception.EndpointServerErrorException;
+import org.alfresco.hxi_connector.common.model.repository.NodeEntry;
+import org.alfresco.hxi_connector.common.util.ErrorUtils;
+import org.alfresco.hxi_connector.prediction_applier.config.NodesApiProperties;
+
+@Component
+@RequiredArgsConstructor
+public class NodesClient extends RouteBuilder
+{
+    public static final String DIRECT_ENDPOINT = "direct:" + NodesClient.class.getSimpleName();
```

Comment:
I think constant name could be more explanatory.
```suggestion
    public static final String NODES_ENDPOINT = "direct:" + NodesClient.class.getSimpleName();
```
or
```suggestion
    public static final String NODES_DIRECT_ENDPOINT = "direct:" + NodesClient.class.getSimpleName();
```